### PR TITLE
LifeCycle_Preview created

### PR DESCRIPTION
**Pull request by Avisheet**
# rmf_obstacle_detector_laserscan waiting to configure

While running  ros2 run rmf_obstacle_detector_laserscan laserscan_detector , I encountered *waiting to configure* .
The Lifecycle_preview provides a detailed description about how to configure and activate the node.

WhatsApp Image 2024-03-16 at 4 58 31 PM

### CODE REVIEW

**Review by Yadunund — CHANGES_REQUESTED** [review]
Appreciate the effort to improve the documentation but there's no need to create a new markdown page summarizing lifecycle nodes. You may add a line to the existing readme that highlights that the nodes are lifecycle node and link to the lifecycle design document. Copy pasting text from existing documentation into a separate file here is not favorable practice.

It's best to add the commands to configure and activate the node right after the existing [command to run the node](https://github.com/open-rmf/rmf_obstacle_detectors#rmf_obstacle_detector_laserscan)

Also please rebase this PR with the latest `main` and ensure there are no unnecessary commits from other contributors.

**Comment by Avisheet** [comment]
Thanks @Yadunund for reviewing and giving me suggestions . I was firstly thinking to provide a different markdown page for this with a brief description .
I will ensure to make the changes as suggested . Can you tell me should I add the picture of this "waiting to configure..." point or should I just add the commands .

**Comment by Yadunund** [comment]
No need to include a screenshot.

File: README.md [hunk 1]
```diff
@@ -14,14 +14,25 @@ Packages that infer and react to the presence of obstacles from sensor inputs.
 
 A ROS 2 node that subscribes to `LaserScan` messages and publishes obstacles to `/rmf_obstacles`.
 
-The node is implemented as an`rclcpp::lifecycle_node` node where upon activation, it calibrates the surroundings based on the range values in the initial few `LaserScan` messages.
+The node is implemented as an[`rclcpp::lifecycle_node`](https://github.com/ros2/demos/tree/rolling/lifecycle) node where upon activation, it calibrates the surroundings based on the range values in the initial few `LaserScan` messages.
```
Comment (Yadunund) [inline README.md:17]:
```suggestion
The node is implemented as an [`rclcpp::lifecycle_node`](https://github.com/ros2/demos/tree/rolling/lifecycle) node where upon activation, it calibrates the surroundings based on the range values in the initial few `LaserScan` messages.
```

**Review by Yadunund — CHANGES_REQUESTED** [review]
Please rebase your PR with `main` and ensure you don't include commits from other authors.

**Comment by Yadunund** [comment]
@Avisheet do you still plan on addressing the comments in the PR?

**Comment by Avisheet** [comment]
Yes @Yadunund . Due to some issues,  I wast able to focus here . But I ensure you that I will be working on this .

**Comment by Avisheet** [comment]
Hey @Yadunund, I have rebased my PR with the main. I request you to review my PR and tell me the desired changes.

**Comment by Yadunund** [comment]
The first commit still has changes from a different author.

**Comment by Avisheet** [comment]
Hey @Yadunund, should I also remove commits 5,6 and 7?

**Comment by Avisheet** [comment]
Hey @Yadunund , the last two commits appeard while resolving the merge errors after removing commits from other authors , can you once review it and tell me the desired changes I need to do now.

**Comment by Yadunund** [comment]
Please resolve conflicts with main

**Review by Yadunund — APPROVED** [review]
Thanks for iterating and sorry for the delay in getting this merged.

**Pull request merged**